src/date-adapters/luxon/index.ts

```diff
@@ -0,0 +1,187 @@
+import { DateTime } from 'luxon';
+import { DateAdapter } from '../date-adapter';
+
+export function adapterFactory(luxon): DateAdapter {
+  /* istanbul ignore next */
+  function coerceDateTime(date: Date | number): DateTime {
+    return typeof date === 'number'
+      ? luxon.DateTime.fromMillis(date)
+      : luxon.DateTime.fromJSDate(date);
+  }
+
+  function addDays(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ days: amount }).toJSDate();
+  }
+
+  function addHours(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ hours: amount }).toJSDate();
+  }
+
+  function addMinutes(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ minutes: amount }).toJSDate();
+  }
+
+  function addSeconds(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ seconds: amount }).toJSDate();
+  }
+
+  function differenceInDays(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('days')
+    );
+  }
+
+  function differenceInMinutes(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('minutes')
+    );
+  }
+
+  function differenceInSeconds(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('seconds')
+    );
+  }
+
+  function endOfDay(date: Date | number): Date {
+    return coerceDateTime(date).endOf('day').toJSDate();
+  }
+
+  function endOfMonth(date: Date | number): Date {
+    return coerceDateTime(date).endOf('month').toJSDate();
+  }
+
+  function endOfWeek(
+    date: Date | number,
+    options?: {
+      weekStartsOn?: number;
+    }
+  ): Date {
+    return coerceDateTime(date)
+      .endOf('week', { useLocaleWeeks: true })
+      .plus({ days: /* istanbul ignore next */ options?.weekStartsOn ?? 0 })
```

Comment:
Should this be `.set({localWeekNumber: options?.weekStartsOn ?? 0})` instead?

https://moment.github.io/luxon/#/intl?id=setting-locale-based-week-data

Reply:
I'm a little bit conflicted on this one. The user can set his week start outside of calendar-utils, which I did for the tests with `luxon.Settings.defaultWeekSettings = {
        firstDay: 7,
        minimalDays: 1,
        weekend: [6, 7],
      };`

So using `options.weekStartsOn` might conflict with the use of `useLocaleWeeks`. How about we do the same as the moment adapter and ignore this option? Instead I'll update the test cases so they'll set the correct `defaultWeekSettings`.

Reply:
> How about we do the same as the moment adapter and ignore this option?

Yes let's do that instead, I think that's the easiest option